lib/cylc/batch_sys_handlers/pbs_multi_cluster.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python2
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2018 NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""PBS batch system job submission and manipulation: multi-cluster variant.
+
+Support PBS clients that front heterogeneous clusters where the Job ID returned
+by qsub is <id>.<server>. PBS 13 qstat and qdel need <id>.<server>@<server>.
+From PBS 14, the standard cylc PBS module works ("@<server>" is not needed).
+
+So this PBS handler writes "job_id@server" to the job status file, and appends
+"@server" to Job IDs returned by qstat, to matched the stored IDs.
+"""
+
+from cylc.batch_sys_handlers.pbs import PBSHandler
+
+
+class PBSMulticlusterHandler(PBSHandler):
+
+    @classmethod
+    def filter_poll_many_output(cls, out):
+        """For job_id's of the form "id.server", return job_id@server."""
+        out = out.strip()
+        job_ids = []
+        lines = out.split('\n')
+        for line in lines[2:]:
+            job = line.split()[0]
+            _, server = job.split('.', 1)
+            job_ids.append(job + '@' + server)
```

Comment:
Can this be written as this:

```python
job_ids.append(cls.manip_job_id(line.split()[0]))
```

Reply:
If really going for it:

```python
def filter_poll_many_output(cls, out):
    return re.sub('(?P<job>.*)\.(?P<host>[^ ]+).*', '\g<job>@\g<host>', out.strip()).splitlines()[2:]
```

Reply:
That will add a dependency on the `re` module. Pure string operations are probably good enough here. (I miss Perl.)

Reply:
@oliver-sanders - I couldn't get your "really going for it" version working, thanks to the vagaries of multi-line regex matching.  Instead went for @matthewrmshin's suggestion plus major tidying.

Reply:
@matthewrmshin - I missed your comment above, earlier, and I have in fact now used `re`.  But that doesn't really matter does it, given that `re` will definitely be loaded in the scheduler program anyway?

> I miss Perl

I don't hear that too often :smiling_imp:

Reply:
No worries. Perl has a few good syntactic sugars that have only started to see daylight in very recent versions of Python.